src/main/java/com/passpnu/passwordmanager/filter/JwtFilter.java

```diff
@@ -0,0 +1,58 @@
+package com.passpnu.passwordmanager.filter;
+
+
+import com.passpnu.passwordmanager.util.JwtUtil;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.AllArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+@AllArgsConstructor
+public class JwtFilter extends OncePerRequestFilter {
+
+    private final UserDetailsService userDetailsService;
+
+    private final JwtUtil jwtUtil;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
+                                    FilterChain filterChain) throws ServletException, IOException {
+
+        final var authHeader = httpServletRequest.getHeader("Authorization");
+
+        String username = null;
+        String jwt = null;
+        final String headerStart = "Bearer ";
+
+        if (authHeader != null && authHeader.startsWith(headerStart)) {
+            jwt = authHeader.substring(headerStart.length());
+            username = jwtUtil.extractUsername(jwt);
+        }
+
+        if (username != null && SecurityContextHolder.getContext().getAuthentication() == null) {
```

Comment:
Checking user on each request by fetching it from DB is not exactly the best idea
You may store basic user details in the token itself (not only the username), it will save you a few requests

Reply:
fixed